doc/devices/fock.rst

```diff
@@ -0,0 +1,95 @@
+
+You can instantiate the Fock device in PennyLane as follows:
+
+.. code-block:: python
+
+    import pennylane as qml
+
+    dev = qml.device('strawberryfields.fock', wires=2, cutoff_dim=10)
+
+The device can then be used just like other devices for the definition and evaluation of QNodes within PennyLane.
+
+For instance, the following simple example defines a :code:`quantum_function` circuit that first displaces
+the vacuum state, applies a beamsplitter, and then returns the photon number expectation.
+This function is then converted into a QNode which is placed on the :code:`strawberryfields.fock` device:
+
+.. code-block:: python
+
+    @qml.qnode(dev)
+    def quantum_function(x, theta):
+        qml.Displacement(x, 0, wires=0)
+        qml.Beamsplitter(theta, 0, wires=[0, 1])
+        return qml.expval(qml.NumberOperator(0))
+
+We can evaluate the QNode for arbitrary values of the circuit parameters:
+
+>>> quantum_function(1., 0.543)
+0.7330132578095255
+
+We can also evaluate the derivative with respect to any parameter(s):
+
+>>> dqfunc = qml.grad(quantum_function, argnum=0)
+>>> dqfunc(1., 0.543)
+1.4660265156190515
+
+The continuous-variable QNodes available via Strawberry Fields can also be combined with qubit-based QNodes
+and classical nodes to build up a hybrid computational model. Such hybrid models can be optimized using
+the built-in optimizers provided by PennyLane.
+
+Device options
+~~~~~~~~~~~~~~
+
+The Strawberry Fields Fock device accepts additional arguments beyond the PennyLane default device arguments.
+
+``cutoff_dim``
+	the Fock basis truncation to be applied when executing quantum functions (``strawberryfields.fock`` only)
+
+``hbar=2``
+	The convention chosen in the canonical commutation relation :math:`[x, p] = i \hbar`.
+	Default value is :math:`\hbar=2`.
+
+``shots=0``
+	The number of circuit evaluations/random samples used to estimate expectation values of observables.
+	The default value of 0 means that the exact expectation value is returned.
+
+	If shots is non-zero, the Fock device calculates the variance of the expectation value(s),
+	and use the `Berry-Esseen theorem <https://en.wikipedia.org/wiki/Berry%E2%80%93Esseen_theorem>`_ to
+	estimate the sampled expectation value.
+
+Supported operations
+~~~~~~~~~~~~~~~~~~~~~
+
+The Strawberry Fields Fock device supports all continuous-variable (CV) operations and observables
+provided by PennyLane, including both Gaussian and non-Gaussian operations:
+
+* **Supported operations:** ``Beamsplitter``, ``ControlledAddition``, ``ControlledPhase``,
+  ``Displacement``, ``Kerr``, ``CrossKerr``, ``QuadraticPhase``, ``Rotation``, ``Squeezing``,
+  ``TwoModeSqueezing``, ``CubicPhase``, ``CatState``, ``CoherentState``, ``FockDensityMatrix``,
+  ``DisplacedSqueezedState``, ``FockState``, ``FockStateVector``, ``SqueezedState``, ``ThermalState``, ``GaussianState``
+
+* **Supported observables:** ``Identity``, ``NumberOperator``, ``X``, ``P``, ``QuadOperator``, ``PolyXP``
```

Comment:
This is cool, but likely to go out of date really quickly 😆 I wonder if there is a way of automating it.

Reply:
Barring automation, it could be cool to display it as an autosummary table with active links,

```rest
:html:`<div class="summary-table">`

.. autosummary::
    :nosignatures:

    ~pennylane.Beamsplitter
    ~pennylane.ControlledAddition
    ~pennylane.ControlledPhase
    ~pennylane.CrossKerr
    ~pennylane.CubicPhase
    ~pennylane.Displacement
    ~pennylane.Interferometer
    ~pennylane.Kerr
    ~pennylane.QuadraticPhase
    ~pennylane.Rotation
    ~pennylane.Squeezing
    ~pennylane.TwoModeSqueezing

:html:`</div>`
```

like we do on the PennyLane operations quickstart page. That way users can click and go directly to the operation docstring. In addition, sphinx will show warnings if any of the names are incorrect/have changed.

Reply:
Cool idea, but how do they become links? Don't we have to crossref to the pennylane repo somehow?

Reply:
Should work out of the box due to intersphinx:
https://github.com/XanaduAI/pennylane-sf/blob/d62e821d387a3e558438f36491d594f7b5a843c6/doc/conf.py#L84

Reply:
Maybe that needs it to be online?

We can push this for now and make the links live in a separate PR if it does not work...

Reply:
Just pulled, works perfectly for me! All links resolve
![image](https://user-images.githubusercontent.com/2959003/79310222-ba617000-7f3a-11ea-85d2-19e99fc13a30.png)

Reply:
Wait, it works now. I did not pull your changes!